src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -398,7 +398,16 @@ public void close(Throwable errorsSoFar)
             Throwable err = Throwables.close(errorsSoFar, obsCloseable, writer, sstableRefs);
 
             if (transaction.isOffline())
+            {
```

Comment:
@jasonstack I had to add this otherwise on CDNB we never publish these metrics

Reply:
FTR, `setEnableMemtableAndCommitLog` is not called on CNDB compactor, thus tracker is considered dummry and txn is offline